[DOC] `FCNClassifier` - add Wang et al reference

itsbharatj opened the pull request:
#### Reference Issues/PRs
Fixes #7862 


#### What does this implement/fix? Explain your changes.
<!--
A clear and concise description of what you have implemented.
-->
Adds Wang et al. as in the references for the FCN classifier docs 


#### Any other comments?
<!--
We value all user contributions, no matter how small or complex they are. If you have any questions, feel free to post
in the dev-chat channel on the sktime discord https://discord.com/invite/54ACzaFsn7. If we are slow to review (>3 working days), likewise feel free to ping us on discord. Thank you for your understanding during the review process.
-->

As pointed out by @fkiraly for checking for the FCN Regressor, I was unable to see their contribution towards it, so I have not added it there.

fkiraly reviewed (changes requested):
Thanks!

* the reference in the text at the top should also be updated, currently it points to reference 1. Reference to 2 should be added, and perhaps 1 should be referred to differently or not at all?
* is there code on GitHub that could have been copied? If so, the GitHub user names should also be credited in the `"authors"` tag.

fkiraly commented:
> is there code on GitHub that could have been copied?

Re that, I remember that code was actually copied, from @hfawaz - we added the credits some time ago.

I also found these
https://github.com/cauchyturing/UCR_Time_Series_Classification_Deep_Learning_Baseline/blob/master/FCN.py
by @cauchyturing

and this:
https://github.com/timeseriesAI/tsai

itsbharatj commented:
Thank you for your comment, @fkiraly. 

> code formatting is failing
According to my understanding of the style, 88 characters should be there, but I had more than that. Fixed comit for this

> Reference to 2 should be added
This is also implemented.

Let me know any additional things you would like me to look on this.

itsbharatj commented:
> is there code on GitHub that could have been copied?

When I compared the code, it did not seemed to be copied. Let me know, if you would like me to further verify this.

Sohaib-Ahmed21 commented:
> @fkiraly can you please run the checks.
> 
> Thank You!

Just for reference, you can check the estimator you changed using `check_estimator` in sktime https://www.sktime.net/en/latest/api_reference/auto_generated/sktime.utils.estimator_checks.check_estimator.html

Sohaib-Ahmed21 commented:
> > @fkiraly can you please run the checks.
> > Thank You!
> 
> Just for reference, you can check the estimator locally you changed using `check_estimator` in sktime https://www.sktime.net/en/latest/api_reference/auto_generated/sktime.utils.estimator_checks.check_estimator.html

Sohaib-Ahmed21 commented:
> > > @fkiraly can you please run the checks.
> > > Thank You!
> > 
> > 
 Just for reference, you can locally check the estimator  changed using `check_estimator` in sktime https://www.sktime.net/en/latest/api_reference/auto_generated/sktime.utils.estimator_checks.check_estimator.html

fkiraly commented:
> @fkiraly can you please run the checks.
> 
> Thank You!

Sure - thank you!

The code formatting is still failing, likely a space or newline somewhere.

fkiraly reviewed (approved):
I fixed it now.
